### DISCUSSION TITLE
Integrating Ubuntu Pro with ESM Repositories into Uyuni

### DISCUSSION THREAD

**Discussion by BoBBer446** [uyuni-project/uyuni#8471] [Q&A]
Hello,

I am currently managing an environment with Ubuntu Pro systems and am looking to integrate these systems into Uyuni for centralized management, specifically focusing on the utilization of Ubuntu's Extended Security Maintenance (ESM) repositories for both `main` and `universe` sources. These are typically represented by the `ubuntu-esm-infra.list` and `ubuntu-esm-apps.list` repositories.

I have Ubuntu Pro systems that I'd like to manage through Uyuni, including the deployment of security updates from the ESM repositories. However, I'm unsure about the best way to integrate these ESM repositories within Uyuni. Here are my specific questions:

1. Is there an existing method or best practice for adding Ubuntu ESM repositories (`ubuntu-esm-infra` for main sources and `ubuntu-esm-apps` for universe sources) to Uyuni?
2. How can I ensure that my Ubuntu Pro systems registered in Uyuni correctly authenticate with and pull updates from these ESM repositories?
3. Are there any considerations or additional steps required to manage Ubuntu Pro systems with ESM repositories in Uyuni that differ from managing standard Ubuntu systems?

I appreciate any guidance, documentation, or insights you can provide on integrating Ubuntu Pro systems with ESM support into the Uyuni management framework.

Thank you for your assistance.

Best regards

**Comment by mcalmer** [comment]
About 1: No, not yet. But in the end you will define the repositories and custom channels and they get mirrored.
About 2: I do not know anything about how the repos authenticate. Please try to find out what mechanisms are used. In case of simple basic auth, you can just add username and password in the repository URL and it should get authenticated correctly.
About 3: As I do not know ESM, I cannot really answer this question. But I would assume "no". You just need to take care that you do not violate the Ubuntu services terms. This is up to every user. Uyuni does not help here.

**Comment by rjmateus** [comment]
Ignoring the terms and conditions mentioned by @mcalmer , after the channels are synchronized you should be able to manage the systems as any other Ubuntu machine.
If you try it, please let us know if authentication worked, and how you set it up. It could be interesting to others in the community.
